core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp

```diff
@@ -267,6 +279,87 @@ VirtualSwapchain::VirtualSwapchain(
                  },
                  this);
 #endif
+
+  const char *const image_dump_path = std::getenv(kImageDumpPathEnv);
+  if (image_dump_path != nullptr) {
+    std::cout << "Swapchain images will be dumped to " << image_dump_path
+              << std::endl;
+    static int frame_count = 1;
+    ImageDumpInfo *dump_info = new ImageDumpInfo{
+        std::string(image_dump_path),
+        width_,
+        height_,
+        _swapchain_info->imageFormat,
+    };
+
+    SetCallback(
```

Comment:
What happens if the user wants to set a callback, but we have this env var set. that means we will end up overwriting it.